json/builder.scm

```diff
@@ -202,7 +225,9 @@ document will be pretty printed.
 Note that when using alists to build JSON objects, symbols or numbers might be
 used as keys and they both will be converted to strings.
 "
-  (json-build scm port escape unicode pretty 0))
+  (if (json-valid? scm)
+      (json-build scm port escape unicode pretty 0)
+      (throw 'xx-json-invalid)))
```

Comment:
shouldn't this be `json-invalid`? i probably need to update the README. Since the builder can throw an exception without the parser argument. Or may be we can rename it to `json-scm-invalid`? So it's clear which one is what. I think I like this option better.

wdyt?